lib/json_validator.rb

```diff
@@ -0,0 +1,123 @@
+# Copyright (c) 2013-2015 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of version 3 of the GNU General Public License as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE about this file by physical or electronic mail,
+# you may find current contact information at www.suse.com
+
+class JsonValidator
+  def initialize(format_version)
+    @global_schema = global_schema(format_version)
+    @scope_schemas = scope_schemas(format_version)
+  end
+
+  def validate(hash)
```

Comment:
It would be good to document the expectations on the hash. Maybe also a more speaking name like `json_hash` would make the code easier to read.